tools/test-backend

```diff
@@ -66,6 +71,56 @@ if __name__ == "__main__":
                       help="Show which tests are slowest.")
 
     (options, args) = parser.parse_args()
+
+    thedir = 'zerver/tests/'
+
+    for suite in args:
+        args[args.index(suite)] = suite.replace("/", ".")
+
+    for suite in args:
+        found_file = False
+        if suite[0].isupper() and "test_" in suite:
+            classname = suite.rsplit('.', 1)[0]
+            for file in os.listdir(thedir):
+                if not file.endswith(".pyc"):
+                    filepath = thedir + file
+                    if not os.path.isdir(filepath):
+                        for line in open(filepath):
+                            if classname in line:
+                                filepath = filepath.replace(".py", ".")
+                                new_suite = filepath + suite
+                                args[args.index(suite)] = new_suite
+                                found_file = True
+                                break
+                if found_file == True:
+                    break
+
+        elif suite[0].isupper():
+            for file in os.listdir(thedir):
+                if not file.endswith(".pyc"):
+                    filepath = thedir + file
+                    if not os.path.isdir(filepath):
+                        for line in open(filepath):
+                            if suite in line:
+                                filepath = filepath.replace(".py", ".")
+                                new_suite = filepath + suite
+                                args[args.index(suite)] = new_suite
+                                found_file = True
+                                break
```

Comment:
It seems like a big chunk of this case is the same as the code in the previous case; can you extract that code into a function so it can be reused?